### PR TITLE
Inventory.payItems()

### PR DESCRIPTION
**Pull request by Birdulon**
## Description
Adds some convenience methods to `Inventory` to handle the common task of checking whether a player can afford to pay a certain cost of items and/or coins/other virtual currencies, and if so, subtracting/removing that cost and proceeding.

Refactors some existing code to make use of these new convenience methods.

## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by Birdulon** [comment]
Tested upgrading and ascending characters and weapons, upgrading relics in-game, didn't test any malicious requests as I'm not set up for that but I think it's good to go.

File: src/main/java/emu/grasscutter/game/managers/InventoryManager.java [hunk 1]
```diff
@@ -38,6 +39,8 @@ public class InventoryManager {
 	
 	private final static int RELIC_MATERIAL_1 = 105002; // Sanctifying Unction
 	private final static int RELIC_MATERIAL_2 = 105003; // Sanctifying Essence
+	private final static int RELIC_MATERIAL_EXP_1 = 2500; // Sanctifying Unction
+	private final static int RELIC_MATERIAL_EXP_2 = 10000; // Sanctifying Essence
```
Comment (ghost) [inline src/main/java/emu/grasscutter/game/managers/InventoryManager.java:43]: Please do not hardcode if you can find the values in the excels.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/game/managers/InventoryManager.java (hunk 1 above)
Comment (Birdulon) [reply src/main/java/emu/grasscutter/game/managers/InventoryManager.java:43]: Can't find them :(

Comment (ghost) [reply src/main/java/emu/grasscutter/game/managers/InventoryManager.java:43]: Its no problem then.

**Review by ghost — APPROVED** [review]

**Pull request merged**